frontend/device/input.lua

```diff
@@ -1208,34 +1214,33 @@ function Input:waitEvent(now, deadline)
                 DEBUG:logEv(event)
                 if event.type == C.EV_KEY then
                     logger.dbg(string.format(
-                        "key event => code: %d (%s), value: %s, time: %d.%06d",
+                        "key event => code: %d (%s), value: %s, time: %s",
                         event.code, self.event_map[event.code] or linux_evdev_key_code_map[event.code], event.value,
-                        event.time.sec, event.time.usec))
+                        time.format_time(event.time)))
                 elseif event.type == C.EV_SYN then
                     logger.dbg(string.format(
-                        "input event => type: %d (%s), code: %d (%s), value: %s, time: %d.%06d",
-                        event.type, linux_evdev_type_map[event.type], event.code, linux_evdev_syn_code_map[event.code], event.value,
-                        event.time.sec, event.time.usec))
+                        "input event => type: %d (%s), code: %d (%s), value: %s, time: %s",
+                        event.type, linux_evdev_type_map[event.type], event.code, linux_evdev_syn_code_map[event.code],
+                        event.value,time.format_time(event.time)))
```

Comment:
Just asking: how worthy/costly is it to do all these `time.format_time()`, here for logger.dbg() that won't show up on devices, and above for all events (are the ev.time always used, or only when needed like for this scrolling thing, in which case isn't it costly ?)

Reply:
One `time.format_time` does a string format (for float) and one multiplication.

In `dbg.lua` one string concatenation is saved.

In `input.lua` two string formats (for int) are saved.

So overall I see almost no cost impact. (In fact the overhead introduced in `UIManager` is saved in `handleTouchEv`.)

About the worthiness of the logged infos @NiLuJe should be asked.

`ev.time` is needed to detect gestures in `gesturedetector.lua`.

Reply:
I'd rather figure out where the non-time one is seeping through, as, yeah, that's a "lot" of overhead when we only ever have a *single* time per event stream (the SYN_REPORT one).

Reply:
(Speaking solely for the common code-path, the debug stuff is entirely harmless as hidden behind an if dbg branch. And I rely on it being that verbose. A lot.).